File: challenges/5-cobaias/php/aryangomes/cobaias.php
<?php

if (count($argv) < 2) {
    throw new \Exception('Informe a quantidade total de testes.');
}

$quantidadeDeCasosDeTestes = $argv[1];

echo 'Digite os casos: ' . PHP_EOL;
echo PHP_EOL;

$casosDeTeste = [];

for ($i = 1; $i <= $quantidadeDeCasosDeTestes; $i++) {

    do {
        $tipoDeCobaia = null;

        $inputCasoDeTeste = readline("Teste $i: ");

        if (!($inputCasoDeTeste)) {
            echo 'Nenhum valor informado! Digite novamente.' . PHP_EOL;
            continue;
        }
        $casoDeTeste = explode(' ', $inputCasoDeTeste);

        $tipoDeCobaia = strtoupper($casoDeTeste[1]);

        $quantidadeDeCobaias = $casoDeTeste[0];
        if (verificaTipoDeCobaia($tipoDeCobaia)) {
            echo 'Tipo de cobaia incorreta! Digite novamente.' . PHP_EOL;
        }
    } while (verificaTipoDeCobaia($tipoDeCobaia));

    if (key_exists($tipoDeCobaia, $casosDeTeste)) {

        $casosDeTeste[$tipoDeCobaia] += (int) $quantidadeDeCobaias;
    } else {
        $casosDeTeste[$tipoDeCobaia] = (int) $quantidadeDeCobaias;
    }
}

$totalDeCasosDeTestesDeCoelhos =  0;

if (key_exists('C', $casosDeTeste)) {
    $totalDeCasosDeTestesDeCoelhos =  $casosDeTeste['C'];
}


$totalDeCasosDeTestesDeRatos = 0;

if (key_exists('R', $casosDeTeste)) {
    $totalDeCasosDeTestesDeRatos =  $casosDeTeste['R'];
}

$totalDeCasosDeTestesDeSapos =   0;

if (key_exists('S', $casosDeTeste)) {
    $totalDeCasosDeTestesDeSapos =  $casosDeTeste['S'];
}

$totalDeCasosDeTestes = $totalDeCasosDeTestesDeCoelhos + $totalDeCasosDeTestesDeRatos +
    $totalDeCasosDeTestesDeSapos;

$percentualTotalDeCasosDeTestesDeCoelhos = number_format((($totalDeCasosDeTestesDeCoelhos / ($totalDeCasosDeTestes / 100))), 2);

$percentualTotalDeCasosDeTestesDeRatos = number_format((($totalDeCasosDeTestesDeRatos / ($totalDeCasosDeTestes / 100))), 2);

$percentualTotalDeCasosDeTestesDeSapos = number_format((($totalDeCasosDeTestesDeSapos / ($totalDeCasosDeTestes / 100))), 2);

echo PHP_EOL;

print_r("Total: $totalDeCasosDeTestes cobaias" . PHP_EOL);

print_r("Total de coelhos: $totalDeCasosDeTestesDeCoelhos" . PHP_EOL);

print_r("Total de ratos: $totalDeCasosDeTestesDeRatos" . PHP_EOL);

print_r("Total de sapos: $totalDeCasosDeTestesDeSapos" . PHP_EOL);

print_r("Percentual de coelhos: $percentualTotalDeCasosDeTestesDeCoelhos %" . PHP_EOL);

print_r("Percentual de ratos: $percentualTotalDeCasosDeTestesDeRatos %" . PHP_EOL);

print_r("Percentual de sapos: $percentualTotalDeCasosDeTestesDeSapos %" . PHP_EOL);

function verificaTipoDeCobaia(?string $tipoDeCobaia): bool
{
    return ($tipoDeCobaia != 'C' &&
        $tipoDeCobaia != 'R' &&
        $tipoDeCobaia != 'S');
}

File: challenges/5-cobaias/java/jesombra/Cobaias.java
import java.util.Scanner;

public class Cobaias{
    public static void main(String[] args) {
        start();
    }

    public static void start(){
        Scanner sc = new Scanner(System.in);
        System.out.print("Digite a quantidade de testes que foram feitos: ");
        int quantidade = sc.nextInt();

        int ratos = 0, sapos = 0, coelhos = 0, totalCobaias = 0;
        for(int i = 0; i < quantidade; i++){
            char tipoCobaias;
            int qtdCobaias;
            System.out.print("Digite a quantidade e o o tipo de cobaia: ");
            qtdCobaias = sc.nextInt();
            tipoCobaias = sc.next().charAt(0);
            totalCobaias += qtdCobaias;
            
            if(tipoCobaias == 'C'){
                coelhos += qtdCobaias;
            } else if (tipoCobaias == 'R'){
                ratos += qtdCobaias;
            } else if (tipoCobaias == 'S'){
                sapos += qtdCobaias;
            }
        }

        System.out.printf("Total: %d cobaias%n", totalCobaias);
        System.out.printf("Total de Coelhos: %d%n", coelhos);
        System.out.printf("Total de Ratos: %d%n", ratos);
        System.out.printf("Total de Sapos: %d%n", sapos);
        System.out.printf("Percentual de Coelhos: %.2f %% %n", ((double) coelhos / totalCobaias * 100));
        System.out.printf("Percentual de Ratos: %.2f %% %n", ((double) ratos / totalCobaias * 100));
        System.out.printf("Percentual de Sapos: %.2f %% %n", ((double) sapos / totalCobaias * 100));

        sc.close();
    }
}